src/test/java/com/cdancy/bitbucket/rest/features/InsightsApiMockTest.java

```diff
@@ -0,0 +1,478 @@
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/repository-not-exist.json")).setResponseCode(404));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final InsightsApi api = baseApi.insightsApi();
+        try {
+
+            final InsightReportPage insightReportPage = api.listReports(projectKey, repoKey, commitHash, 100, 0);
+            assertThat(insightReportPage).isNotNull();
+            assertThat(insightReportPage.values()).isEmpty();
+            assertThat(insightReportPage.errors()).isNotEmpty();
+
+            final Map<String, ?> queryParams = ImmutableMap.of(limitKeyword, 100, startKeyword, 0);
+            assertSent(server,
+                       getMethod,
+                       restApiPath + BitbucketApiMetadata.API_VERSION
+                           + projectsKeyword + projectKey + repoKeyword + repoKey + commitKeyword + commitHash + reportsKeyword, queryParams);
+        } finally {
+            baseApi.close();
+            server.shutdown();
+        }
+    }
+
+    public void testGetReport() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/insight-report.json")).setResponseCode(200));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final InsightsApi api = baseApi.insightsApi();
+        try {
+
+            final String reportKey = qwertyKeyword;
+            final InsightReport report = api.getReport(projectKey, repoKey, commitHash, reportKey);
+            assertThat(report).isNotNull();
+            assertThat(report.key()).isEqualTo(reportKey);
+
+            assertSent(server,
+                       getMethod,
+                       restApiPath + BitbucketApiMetadata.API_VERSION
+                           + projectsKeyword + projectKey + repoKeyword + repoKey + commitKeyword + commitHash + reportsKeyword + "/" + reportKey);
+        } finally {
+            baseApi.close();
+            server.shutdown();
+        }
+    }
+
+    public void testGetReportOnError() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/insight-report-error.json")).setResponseCode(404));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final InsightsApi api = baseApi.insightsApi();
+        try {
+
+            final String reportKey = qwertyKeyword;
+            final InsightReport report = api.getReport(projectKey, repoKey, commitHash, reportKey);
+            assertThat(report).isNotNull();
+            assertThat(report.errors()).isNotEmpty();
+
+            assertSent(server,
+                       getMethod,
+                       restApiPath + BitbucketApiMetadata.API_VERSION
+                           + projectsKeyword + projectKey + repoKeyword + repoKey + commitKeyword + commitHash + reportsKeyword + "/" + reportKey);
```

Comment:
:warning: Line is longer than 140 characters (found 147). (#QR118:154#)-CHECKSTYLE_LineLengthCheck